go-selinux/selinux_linux.go

```diff
@@ -1066,6 +1068,18 @@ func chcon(fpath string, label string, recurse bool) error {
 		return setFileLabel(fpath, label)
 	}
 
+	if lazy {
+		currentLabel, err := fileLabel(fpath)
```

Comment:
@haircommander do we have a guarantee that the top level will always be labeled last? Especially with `pwalkdir`.

Reply:
not yet, though @kolyshkin is going to work on such an enhancement

Reply:
Please see https://github.com/opencontainers/selinux/pull/161